Assets/Scripts/TorchManager.cs

```diff
@@ -7,6 +7,8 @@ public class TorchManager : MonoBehaviour
     [SerializeField] private bool isTorchOn = false;
     [SerializeField] private Light TorchLight;
 
```

Comment:
Now when you removed interaction logic from player you should put it there.
Assign this script as interactable by adding inkeritance from interface
`public class TorchManager : MonoBehaviour, IInteractable`
This will force you to implement method form this so do it.
```cs
public void Interact(GameObject interacter)
{
  var playerHands  = interacter.GetComponent<PlayerHends>();
   // Copied form you script
   if (!playerHands.IsHoldingTorch)
   {
       playerHands.TakeTorch(IsTorchOn);
       Take();
   }
}
```